src/overseers/overseer.py

```diff
@@ -66,33 +85,53 @@ def mytardis_setup(self) -> IntrospectionConfig:
         """Getter for mytardis_setup. Sends API request if self._mytardis_setup is None"""
         return self._mytardis_setup or self.get_mytardis_setup()
 
-    @staticmethod
-    def resource_uri_to_id(uri: URI) -> int:
-        """Gets the id from a resource URI
+    def check_identifiers_enabled_for_type(self, object_type: MyTardisObject) -> None:
+        """Check if identifiers are enabled for the given object type.
 
-        Takes resource URI like: http://example.org/api/v1/experiment/998
-        and returns just the id value (998).
+        Raises AssertionError if identifiers are not enabled for the given object type.
+        """
 
-        Args:
-            uri: str - the URI from MyTardis
+        assert (
```

Comment:
Nice that resolves a lot of the duplications. Bandit complains about raw Asserts due to them not working when compiled as optimized tho, so worth changing to explicitly raising errors (poteintally just to shut bandit up on future PRs).